### ISSUE TITLE
Inference code giving incorrect output (not able to identify reason?)

### ISSUE THREAD

**Issue by akshayjaikanth** [LangDaniel/MAEMI#4]
I was using three subjects and trying to run the model with 10 epochs and assumed that the model would overfit on those three subjects. In the par.yml file I passed  the mae_visualise_vit_base.pth in the resume parameter and trained the model 

In the inference function I assumed the output of the mae would be the reconstructed image and tried outputting but the output does not seem to match the input . Is it due to underfitting or am I making some mistake?

 I have attached a sample image below

I modified the compute_anomaly function to plot this and have attached the code snippet below
        y_raw, _ = self.forward(img)
        y = y_raw.squeeze()

        diff = (img.squeeze() - y.squeeze())**2
        for ii in range(diff.shape[0]):
            diff[ii] = self.filtr(diff[ii])
        plt.figure(figsize=(10, 5))

        
        plt.subplot(1, 3, 1)
        plt.imshow(y.squeeze()[0, :, :, 0], cmap='gray')
        plt.title('Reconstructed Image')
        plt.axis('off')

        plt.subplot(1, 3, 2)
        plt.imshow(diff[0, :, :, 0], cmap='gray')
        plt.title('Difference Image')
        plt.axis('off')

        plt.subplot(1, 3, 3)
        plt.imshow(img[0, :, :, 0], cmap='gray')
        plt.title('Actual Image')
        plt.axis('off')

![Image](https://github.com/user-attachments/assets/e16817f1-a054-49c8-a4cd-57def4c5958b)

**Comment by LangDaniel** [comment]
Looking at the reconstructed images you provided it's evident that the MAE did not learn properly.
Examples of how the reconstructions should look like are provided in the paper.

There can be small differences depending on the data and hyperparameter settings,
but your reconstructions do not resemble the desired output at all.
Unfortunately, I cannot tell you from afar what is going wrong.
Maybe your model is not overfitting properly?
However, this has to do something with the training strategy and not the model architecture.
